File: src/main/java/net/atos/client/zgw/ztc/util/ZaakTypeUtil.java
package net.atos.client.zgw.ztc.util;

import java.time.LocalDate;
import java.time.Period;

import net.atos.client.zgw.ztc.model.generated.ZaakType;

public class ZaakTypeUtil {
    private ZaakTypeUtil() {
    }

    public static boolean isServicenormBeschikbaar(ZaakType zaakType) {
        return zaakType.getServicenorm() != null &&
                !Period.parse(zaakType.getServicenorm()).normalized().isZero();
    }

    public static boolean isNuGeldig(ZaakType zaakType) {
        final LocalDate eindeGeldigheid = zaakType.getEindeGeldigheid();
        return zaakType.getBeginGeldigheid().isBefore(
                LocalDate.now().plusDays(1)
        ) && (eindeGeldigheid == null || eindeGeldigheid.isAfter(LocalDate.now()));
    }
}

File: src/main/kotlin/net/atos/client/zgw/ztc/util/ZaakTypeUtil.kt
package net.atos.client.zgw.ztc.util

import net.atos.client.zgw.ztc.model.generated.ZaakType
import java.time.LocalDate
import java.time.Period

fun isServicenormBeschikbaar(zaakType: ZaakType): Boolean {
    return zaakType.servicenorm != null &&
        !Period.parse(zaakType.servicenorm).normalized().isZero
}

fun isNuGeldig(zaakType: ZaakType): Boolean {
    val eindeGeldigheid = zaakType.eindeGeldigheid
    return zaakType.beginGeldigheid.isBefore(
        LocalDate.now().plusDays(1)
    ) && (eindeGeldigheid == null || eindeGeldigheid.isAfter(LocalDate.now()))
}
